cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -174,22 +181,46 @@ async def _batch_async(
 
         return await asyncio.wait_for(asyncio.gather(*tlm_query_tasks), timeout=timeout)  # type: ignore[arg-type]
 
-    def prompt(self, prompt: str, options: Optional[TLMOptions] = None) -> TLMResponse:
+    def prompt(
+        self,
+        prompt: Union[str, Collection[str]],
+        options: Union[None, TLMOptions, Collection[Union[TLMOptions, None]]] = None,
+        timeout: Optional[float] = None,
+        retries: int = 1,
+    ) -> Union[TLMResponse, List[TLMResponse]]:
         """
         Get response and confidence from TLM.
 
         Args:
-            prompt (str): prompt for the TLM
-            options (Optional[TLMOptions]): options to parameterize TLM with. Defaults to None.
+            prompt (str): prompt (or list of prompts) for the TLM
+            options (None | TLMOptions | Collection[TLMOptions |  None], optional): collection of options (or instance of options) to pass to prompt method. Defaults to None.
```

Comment:
I also just noticed `model` is part of TLMOptions? 
We definitely shouldn't allow somebody to switch models in the middle of a batch. `model` was intended to be part of the TLM initialization args, not part of the prompt() args:

https://www.notion.so/cleanlab/Finalest-TLM-API-f376c74140e84326b2daef4d20bb7822

@ulya-tkch Are you able to fix this in a future PR?

Reply:
> this is a bit unclear to me, why would somebody pass in collection of options.

Not sure there's a ton of utility now, but as we add more `TLMOptions`, this could be useful for users. I don't feel super strongly about it, so can remove if you prefer.

> We definitely shouldn't allow somebody to switch models in the middle of a batch. model was intended to be part of the TLM initialization args, not part of the prompt() args:

I can add `model` to the TLM initialization. Is there _also_ an issue with just having it in `TLMOptions` as well (especially if we limit one `TLMOptions` per batch)? I'm not sure I see the problem w/ letting users override it here.

Reply:
IMO things that can be natural for people to change in runs should be kwargs to `prompt` and `get_confidence_score` methods. 

Things that are unnatural for people to change in runs should be declared in initialization to TLM only. 

Let's not make this hold up this time-sensitive PR any further.